template.rb

```diff
@@ -279,16 +279,18 @@ def gem_exists?(name)
     end
   end
 
+  rails_command "db:create"
```

Comment:
You won't have a chance to edit the database.yml to add credentials if we automatically do this. That's one reason why it isn't run automatically.

Reply:
ah-ha, I assumed there was a reason why that was left out.  I'll remove that and update the instructions to just create the DB before the noticed generator so the instructions will work as-is and will still need just 1 migration

Reply:
Oh yeah, now I remember. The noticed generator looks at the database to determine which json column type to use. Maybe there's a better way to do that without having to load active record?

Reply:
Or... We could have several different generators and have you run the one e for your database.

Reply:
That sounds like a good option and will dig into that a little more when I have some time 🤔  .

Reply:
Damn, so I tried this and the Noticed generator will always have to query the database connection to see if `json` is a valid type.

https://github.com/rails/rails/blob/c88ca3b72083449fd242ed5aaf1b557f88a9e7b2/railties/lib/rails/generators/generated_attribute.rb#L63-L66

Reply:
Ah, that's a bummer.  If there are more options for updating this, I don't see them being as straightforward as updating the order of those commands like I was hoping for.

Should I close this PR out for the time being and revisit sometime in the future with some more options?

Reply:
Yeah, probably. 😩

We could skip the Rails model generator and implement it from scratch, but it's not really the most maintainable option. I'm not sure of any way to improve it either. This is one of the reasons why Rails generators are hard.

That said, I did learn about this method, so we could probably refactor the model generator to use this method from Rails and remove our db adapter checking code.

Reply:
Yeah, that seems like it could be handy in the future.  That could be a fun task, but the underlying issue is more of an inconvenience than a bug.  Thanks for taking a look gonna close this for now.